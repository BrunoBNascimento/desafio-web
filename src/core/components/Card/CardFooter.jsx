import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const CardFooterWrapper = styled.section`
  width:60%;
  padding:10px 0 0 0;
  margin-top: -10px;
  display:flex;
  section{
    padding:0 10px;
    color:#D89200;
    &:first-child{
      padding:0 10px 0 0;
    }
  }
`;

const CardFooter = ({ children }) => (
  <CardFooterWrapper>
    {children}
  </CardFooterWrapper>
);

CardFooter.propTypes = {
  children: PropTypes.node.isRequired,
};

export default CardFooter;
