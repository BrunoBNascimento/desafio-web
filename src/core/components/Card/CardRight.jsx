import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';


const Styled = styled.section`
  width:30%;
`;

const CardRight = ({ children }) => (
  <Styled>
    {children}
  </Styled>
);

CardRight.propTypes = {
  children: PropTypes.node.isRequired,
};

export default CardRight;
