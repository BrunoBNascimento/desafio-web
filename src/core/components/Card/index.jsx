import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const CardWrapper = styled.section`
  overflow-wrap: break-word;
  height:auto;
  padding: 20px 0;
  border-bottom: 1px solid #E0E0E0;
  display:flex;
  flex-flow :wrap;
  justify-content: space-between;
  &:first-child{
    padding:30px 0 20px 0;
  }
`;

const CardContent = styled.div`
  max-width:${props => props.contentSize}%;
`;

const Card = ({
  title,
  body,
  children,
  contentSize,
}) => (
  <CardWrapper>
    <CardContent contentSize={contentSize}>
      <h1 className="sub-title blue-text">{title}</h1>
      <p className="card-body">{body}</p>
    </CardContent>
    {children}
  </CardWrapper>
);

Card.defaultProps = {
  children: '',
  contentSize: 65,
};

Card.propTypes = {
  title: PropTypes.string.isRequired,
  body: PropTypes.string.isRequired,
  children: PropTypes.node,
  contentSize: PropTypes.number,
};

export default Card;
