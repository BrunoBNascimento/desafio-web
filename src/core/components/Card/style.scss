.card-content{
  max-width:65%;
}