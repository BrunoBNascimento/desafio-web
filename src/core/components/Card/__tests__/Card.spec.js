import React from 'react';
import enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Card from '../';

// tslint:disable-next-line:no-any
enzyme.configure({ adapter: new Adapter() });

describe('Card', () => {
  it('should render Card', () => {
    const card = shallow(<Card title="xpto" body="xpto" contentSize={100} />);
    expect(card).toMatchSnapshot();
  });
});
