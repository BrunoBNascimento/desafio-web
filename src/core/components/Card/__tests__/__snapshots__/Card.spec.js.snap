// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render Card 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    body="xpto"
    contentSize={100}
    title="xpto"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.div
          contentSize={100}
>
          <h1
                    className="sub-title blue-text"
          >
                    xpto
          </h1>
          <p
                    className="card-body"
          >
                    xpto
          </p>
</styled.div>,
        "",
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <h1
              className="sub-title blue-text"
>
              xpto
</h1>,
            <p
              className="card-body"
>
              xpto
</p>,
          ],
          "contentSize": 100,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "xpto",
              "className": "sub-title blue-text",
            },
            "ref": null,
            "rendered": "xpto",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "xpto",
              "className": "card-body",
            },
            "ref": null,
            "rendered": "xpto",
            "type": "p",
          },
        ],
        "type": [Function],
      },
      "",
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.div
            contentSize={100}
>
            <h1
                        className="sub-title blue-text"
            >
                        xpto
            </h1>
            <p
                        className="card-body"
            >
                        xpto
            </p>
</styled.div>,
          "",
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <h1
                className="sub-title blue-text"
>
                xpto
</h1>,
              <p
                className="card-body"
>
                xpto
</p>,
            ],
            "contentSize": 100,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "xpto",
                "className": "sub-title blue-text",
              },
              "ref": null,
              "rendered": "xpto",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "xpto",
                "className": "card-body",
              },
              "ref": null,
              "rendered": "xpto",
              "type": "p",
            },
          ],
          "type": [Function],
        },
        "",
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
